AI-generated Systems

domalessi opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do? What is the motivation?
Adds a page to describe the new "AI-generated Systems" functionality for IDP/Software Catalog. 

### Merge instructions

<!-- 
If you're waiting for a release or there are other considerations that you want us to be aware of, list them here. 
If the PR is ready to be merged once it receives the required reviews, check the box below after you've created the PR.
-->

Merge readiness:
- [ ] Ready for merge

**For Datadog employees**:

Your branch name MUST follow the `<name>/<description>` convention and include the forward slash (`/`). Without this format, your pull request will not pass CI, the GitLab pipeline will not run, and you won't get a branch preview. Getting a branch preview makes it easier for us to check any issues with your PR, such as broken links.

If your branch doesn't follow this format, rename it or create a new branch and PR.

[6/5/2025] Merge queue has been disabled on the documentation repo. If you have write access to the repo, the PR has been reviewed by a Documentation team member, and all of the required checks have passed, you can use the **Squash and Merge** button to merge the PR. If you don't have write access, or you need help, reach out in the #documentation channel in Slack.

### Additional notes
<!-- Anything else we should know when reviewing?-->

<!-- Previewing the PR: Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running. -->

janine-c commented on content/en/internal_developer_portal/software_catalog/entity_model/ai-generated-systems.md:
```diff
@@ -0,0 +1,70 @@
+2. In the **Suggested Updates** section, click **Enable System Recommendations**.
+
+{{< img src="path/to/your/image-name-here.png" alt="Your image description" style="width:100%;" >}}
+
```
```suggestion
```

domalessi replied:
Leaving for now just so I don't forget we want an image here once the UI is ready!

janine-c commented on content/en/internal_developer_portal/software_catalog/entity_model/ai-generated-systems.md:
```diff
@@ -0,0 +1,70 @@
+
+To enable:
+
+1. Navigate to **IDP** > **Software Catalog** > **Suggestions**.
```
Could consider linking to https://app.datadoghq.com/software?selectedComponent=suggestionsTable here and in other places where we tell users to go into Datadog 🙂

janine-c reviewed (commented):
Super cool feature!! Just want to make sure we don't accidentally publish with these image placeholders, but will be happy to approve once they're out!

janine-c commented:
I totally forgot to mention this in my review, but I see that this has a failing link check in GitLab. It looks like it might be from a different PR - in that case, I think that merging in master should help 🙂

domalessi commented on content/en/internal_developer_portal/software_catalog/entity_model/ai-generated-systems.md:
```diff
@@ -0,0 +1,68 @@
+- Description
+- Components
+
+After accepting a System, you can update additional metadata on the entity page in your Catalog. Accepted Systems are automatically tagged with `created_by:ai` so you can identify which Systems in your Catalog were originally created by AI.
```
```suggestion
After accepting a System, you can update additional metadata on the entity page in your Catalog. AI-generated Systems are automatically tagged with `created_by:ai` so you can identify which Systems in your Catalog were originally created by AI.
```

domalessi commented:
@janine-c @roxanne-moslehi Alright, team, I think we're all set! I've updated the placeholder images with screenshots from demo org. I've also addressed all comments left by each of you :) Thank you v much for all the feedback!

janine-c commented on content/en/internal_developer_portal/software_catalog/entity_model/ai-generated-systems.md:
```diff
@@ -28,22 +28,19 @@ Recommendations include service, datastore, and queue entity types and automatic
+AI-generated Systems must be enabled once per organization by a user with the `org_management` permission (included in the default [Datadog Admin role][1]).
 
-To enable:
+To enable, navigate to **IDP** > **Software Catalog** > [**Suggestions**][2], and click **Enable System Recommendations**.
```
```suggestion
To enable it, navigate to **IDP** > **Software Catalog** > [**Suggestions**][2], and click **Enable System Recommendations**.
```

janine-c commented on content/en/internal_developer_portal/software_catalog/entity_model/ai-generated-systems.md:
```diff
@@ -0,0 +1,68 @@
+
+## Overview
+
+AI-generated Systems surface recommended System entities that you can add to your Software Catalog. Bits AI generates these recommendations using:
```
I just noticed that in your screenshot, the UI says that it's "Bits AI" making the recommendations. I wonder if that would make more sense than saying that Systems recommend System entities, which, although probably true, might feel a little more ambiguous?

janine-c commented on content/en/internal_developer_portal/software_catalog/entity_model/ai-generated-systems.md:
```diff
@@ -0,0 +1,68 @@
+
+{{< img src="/tracing/internal_developer_portal/enable-system-recs.png" alt="The Suggestions page in Software Catalog, showing an option to enable System Recommendations" style="width:100%;" >}}
+
+Recommendations may take up to 30 minutes to appear after enablement.
```
Maybe consider putting this above the image? Otherwise, I think it might be easy to miss the single line of text below it.

janine-c commented on content/en/internal_developer_portal/software_catalog/entity_model/ai-generated-systems.md:
```diff
@@ -0,0 +1,68 @@
+---
+title: AI-generated Systems
```
I just found some guidance on [hyphens and capitalization](https://datadoghq.atlassian.net/wiki/spaces/WRITING/pages/5341577434/Capitalization#Capitalization-with-hyphens), which makes it look like this should be "AI-Generated."

janine-c reviewed (approved):
It's a thing of beauty, Dom!! I left some very minor suggestions, but no showstoppers. I can't wait to see this go live!

janine-c commented:
Also @domalessi just tagging you to make sure you saw my comment above - I think merging in master should fix the broken link preview check so you can merge!

roxanne-moslehi commented on content/en/internal_developer_portal/software_catalog/entity_model/ai-generated-systems.md:
```diff
@@ -0,0 +1,68 @@
+
+To enable, navigate to **IDP** > **Software Catalog** > [**Suggestions**][2], and click **Enable System Recommendations**.
+
+{{< img src="/tracing/internal_developer_portal/enable-system-recs.png" alt="The Suggestions page in Software Catalog, showing an option to enable System Recommendations" style="width:100%;" >}}
```
Slightly updated screenshot to use

<img width="1423" height="804" alt="image" src="https://github.com/user-attachments/assets/18cf4cfd-9a9e-4b4f-b17b-1ecea4a875e0" />

domalessi commented on content/en/internal_developer_portal/software_catalog/entity_model/ai-generated-systems.md:
```diff
@@ -0,0 +1,70 @@
+Bits AI analyzes the following data sources to identify candidate Systems:
+- Observability data, including distributed traces
+- Component dependencies (automatically detected with APM or manually declared)
+- Ownership metadata from your Software Catalog
```
```suggestion
Bits AI recommends AI-generated Systems that you can add to your Software Catalog. It analyzes the following data sources to identify candidate Systems:
- Observability data, including distributed traces
- Component dependencies (automatically detected with APM or manually declared)
- Ownership metadata from your Software Catalog
```

domalessi commented on content/en/internal_developer_portal/software_catalog/entity_model/ai-generated-systems.md:
```diff
@@ -0,0 +1,68 @@
+  tag: "Documentation"
+  text: "Discover Entities"
+---
+
```
```suggestion

{{< callout url="https://www.datadoghq.com/product-preview/idp-preview-features/" d_target="#signupModal" btn_hidden="false" header="Join the Preview for AI-generated Systems!" >}}
{{< /callout >}}

```

domalessi commented on content/en/internal_developer_portal/software_catalog/entity_model/entity_types.md:
```diff
@@ -44,6 +44,8 @@ In Software Catalog, you can collect non-instrumented services by declaring them
 In Software Catalog, a system (`kind:system`) is a group of entities that cooperate to perform a broader function. For example, you can group multiple instrumented APM services into a system because they are operated by the same team. You can also use `system` to represent a full microservice-based architecture, and include components like APIs, datastores, queues, and other common building block ...
 
+Systems can be manually defined or added through [AI-generated Systems][2].
+
```
Fair. I removed that line!
